src/main/java/com/waytta/SaltAPIBuilder.java

```diff
@@ -47,515 +44,561 @@
+        saltFunc.put("expr_form", targettype);
+        saltFunc.put("fun", myfunction);
+        addArgumentsToSaltFunction(myarguments, saltFunc);
+        addKwArgumentsToSaltFunction(mykwarguments, saltFunc);
+
+        return saltFunc;
+    }
+
+    private void addKwArgumentsToSaltFunction(String mykwarguments, JSONObject saltFunc) {
+        if (mykwarguments.length() > 0) {
+            Map<String, String> kwArgs = new HashMap<String, String>();
+            // spit on comma seperated not inside of quotes
+            String[] kwargItems = mykwarguments.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)");
+            for (String kwarg : kwargItems) {
+                // remove spaces at begining or end
+                kwarg = kwarg.replaceAll("^\\s+|\\s+$", "");
+                kwarg = kwarg.replaceAll("\"|\\\"", "");
+                if (kwarg.contains("=")) {
+                    String[] kwString = kwarg.split("=");
+                    if (kwString.length > 2) {
+                        // kwarg contained more than one =. Let's put the string
+                        // back together
+                        String kwFull = new String();
+                        for (String kwItem : kwString) {
+                            // Ignore the first item as it will remain the key
+                            if (kwItem == kwString[0])
+                                continue;
+                            // add the second item
+                            if (kwItem == kwString[1]) {
+                                kwFull += kwItem;
+                                continue;
+                            }
+                            // add all other items with an = to rejoin
+                            kwFull += "=" + kwItem;
+                        }
+                        kwArgs.put(kwString[0], kwFull);
+                    } else {
+                        kwArgs.put(kwString[0], kwString[1]);
+                    }
+                }
+            }
+            // Add any kwargs to json message
+            saltFunc.element("kwarg", kwArgs);
+        }
+    }
+
+    private void addArgumentsToSaltFunction(String myarguments, JSONObject saltFunc) {
+        if (myarguments.length() > 0) {
+            List<String> saltArguments = new ArrayList<String>();
+            // spit on comma seperated not inside of quotes
+            String[] argItems = myarguments.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)");
+
+            for (String arg : argItems) {
+                // remove spaces at begining or end
+                arg = arg.replaceAll("^\\s+|\\s+$", "");
+                arg = arg.replaceAll("\"|\\\"", "");
+                saltArguments.add(arg);
+            }
+
+            // Add any args to json message
+            saltFunc.element("arg", saltArguments);
+        }
+    }
 
     /*
      * We'll use this from the <tt>config.jelly</tt>.
      */
     public String getServername() {
-	return servername;
```

Comment:
Let's leave the get methods up above the constructor. There's a reformatting I'd like to see happen that removes the arguments from the constructor in favor of set calls. If/when that goes through, these should all be near the @DataBoundConstructor